### PR TITLE
Message Bus Module

### PR DESCRIPTION
**Pull request by rob5standingby**
## Why?
- Adds a new module, `Bus`, to the SDK in order to do CRUD operations on Message Bus Channels

## What changed?
- New Module `Bus` added, with submodule `Channels`
- `Channels` submodule has the following functions
  - `create`
  - `delete`
  - `get`
  - `update`
- Added tests

### CODE REVIEW

File: src/bus/channels.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+   *   .create({
+   *     name: 'Channel 46',
+   *     organizationId: '28cc036c-d87f-4f06-bd30-1e78c2701064',
+   *     serviceId: 'abc123service'
+   *   })
+   *   .then((channel) => console.log(channel))
+   *   .catch((err) => console.log(err));
+   */
+  create(channel = {}) {
+    const requiredFields = ['name', 'organizationId', 'serviceId'];
```
Comment (zachary-kuhn) [inline src/bus/channels.js]: Can we look into `util`'ing this pattern up? We use it so many places!

File: src/config/audiences.js [hunk 2]
```diff
@@ -1,4 +1,14 @@
 export default {
+  bus: {
+    production: {
+      clientId: 'bmoJ2jhfyIP5s4IZx5Ss6xQ9lmkoDixO',
+      host: 'https://bus.ndustrial.io'
+    },
+    staging: {
+      clientId: '8qY2xJob1JAxhmVhIDLCNnGriTM9bct8',
```
Comment (zachary-kuhn) [inline src/config/audiences.js]: Oh shoot, this ID I gave you is the ID used by Contxt coordinator's staging instance. The client ID of the staging instance is `1HD1NG1VTBtkqRt2HRRj3E3hdmqmwzoz`

File: src/bus/channels.js (hunk 1 above)
Comment (rob5standingby) [reply src/bus/channels.js]: A helper function may good to put into `contxt-express`

File: src/config/audiences.js (hunk 2 above)
Comment (rob5standingby) [reply src/config/audiences.js]: Got it. Thx! :D

File: src/bus/channels.js (hunk 1 above)
Comment (rob5standingby) [reply src/bus/channels.js]: Oh wait, i'm not in an API project... I forgot what I was reading. yes some sort of util function would be nice

**Comment by rob5standingby** [comment]
@zachary-kuhn Updated the client id!

**Comment by zachary-kuhn** [comment]
Hey! Also make sure to bump that version.

File: src/bus/channels.spec.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+        request = {
+          ...baseRequest,
+          post: this.sandbox.stub().resolves(channelFromServerBeforeFormat)
+        };
+
+        const channels = new Channels(baseSdk, request);
+        channels._baseUrl = expectedHost;
+        console.log('channels is');
+        console.log(channels);
+        console.log(channels.create);
```
Comment (greg5green) [inline src/bus/channels.spec.js]: Can you remove these console.logs?

Comment (rob5standingby) [reply src/bus/channels.spec.js]: Oof~

Comment (rob5standingby) [reply src/bus/channels.spec.js]: Done!

File: src/bus/channels.spec.js [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+      beforeEach(function() {
+        expectedChannelId = faker.random.uuid();
+        channelFromServerAfterFormat = fixture.build('channel', {
+          id: expectedChannelId
+        });
+        expectedOrganizationId = channelFromServerAfterFormat.organizationId;
+        expectedServiceId = channelFromServerAfterFormat.serviceId;
+        channelFromServerBeforeFormat = fixture.build(
+          'channel',
+          { id: expectedChannelId },
```
Comment (greg5green) [inline src/bus/channels.spec.js]: Why not pass in channelFromServerAfterFormat?

Comment (rob5standingby) [reply src/bus/channels.spec.js]: Ok

Comment (greg5green) [reply src/bus/channels.spec.js]: You can pull `expectedChannelId` from `channelFromServerAfterFormat` too

Comment (rob5standingby) [reply src/bus/channels.spec.js]: Ah yeah that makes it easier to read

File: src/bus/channels.spec.js [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+      let request;
+      let promise;
+
+      beforeEach(function() {
+        channelFromServerBeforeFormat = fixture.build('channel', null, {
+          fromServer: true
+        });
+        channelToServerAfterFormat = fixture.build(
+          'channel',
+          channelFromServerBeforeFormat,
```
Comment (greg5green) [inline src/bus/channels.spec.js]: This should be channelToServerBeforeFormat?

**Comment by rob5standingby** [comment]
@greg5green Let me know if you see anything else. Addressed those comments

**Comment by greg5green** [comment]
:shipit:

**Pull request merged**